### PR TITLE
[8028] execd as wazuh module

### LINKED ISSUE #8028: Execd as wazuh module
|Branch|
|---|
|[8028-execd-as-wazuh-module](https://github.com/wazuh/wazuh/tree/8028-execd-as-wazuh-module)|

## Description
This issue points to the unification of processes on the agent side, in which it aims to only have wazuh-agentd and wazuh-modulesd.

One of the biggest challenges here is to decompose the main of the old module, separating what is the initialization, from the main processing loop, as well as the rearrangement of the configuration reading.

For this type of modification, the main thing is to normalize the module, to maintain the structure and design of wazuh-modulesd, as it is used in the pre-existing modules.

## DoD
- [x] Remove process from installation.
- [x] Add to makefile wazuh-modulesd rule, the dependency of execd compilation.
- [x] Expose methods (Mainloop, read config, communication, shutdown).
- [x] Fix and update logs so that the logging looks like: `wazuh-modulesd:execd`
- [x] Normalize configuration and read this.
- [x] Initialize module and call the main loop.
- [x] Update installers as needed.
- [x] Start Wazuh module in the server, independently of Wazuh configuration.
- [x] Initialize argument received in wmodule calls. --  context->data
- [x] execd queue re-organization.
- [x] Functional tests (Windows and Linux)
- [x] PR CI Pass (Only Build and unit-tests)

## Affected repositories
- Wazuh-jenkins: Issue [#2479](https://github.com/wazuh/wazuh-jenkins/issues/2479)
- Wazuh-QA: Issue [#1204](https://github.com/wazuh/wazuh-qa/issues/1204)
- Wazuh-Documentation: Issue [#3656](https://github.com/wazuh/wazuh-documentation/issues/3656)

### PR DESCRIPTION
**Pull request by tinachoB**
|Related issue|
|---|
|Closes #8028 |

## **Description**
This issue points to the unification of processes on the agent side, in which it aims to only have wazuh-agentd and wazuh-modulesd.

One of the biggest challenges here is to decompose the main of the old module, separating what is the initialization, from the main processing loop, as well as the rearrangement of the configuration reading.

For this type of modification, the main thing is to normalize the module, to maintain the structure and design of wazuh-modulesd, as it is used in the pre-existing modules.

## **DoD**
- [x] Remove process from installation.
- [x] Add to makefile wazuh-modulesd rule, the dependency of execd compilation.
- [x] Expose methods (Mainloop, read config, communication, shutdown).
- [x] Normalize configuration and read this.
- [x] Initialize module and call the main loop.
- [x] Fix unit tests
- [ ] PR CI Pass

### CODE REVIEW

File: src/shared/wmodules_health_status.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                position = 0;
+                break;
+            }
+        } else {
+            position += recv_ret;
+        }
+    }
+    msg[position] = '\0';
+    if (position > 0) {
+        const char *jsonErrPtr;
```
Comment (konangc87) [inline src/shared/wmodules_health_status.c:65]: unused variable.

File: src/wazuh_modules/wm_execd.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+#include "sym_load.h"
+#include "defs.h"
+#include "mq_op.h"
+#include "../os_net/os_net.h"
+
+static void* wm_execd_main(wm_execd_t *data);        // Module main function. It won't return
+static void wm_execd_destroy(wm_execd_t *data);      // Destroy data
+const char *WM_EXECD_LOCATION = "execd";             // Location field for event sending
+cJSON *wm_execd_dump(const wm_execd_t *data);
+int wm_execd_message(const char *data);
```
Comment (konangc87) [inline src/wazuh_modules/wm_execd.c:26]: not needed.

**Review by konangc87 — CHANGES_REQUESTED** [review]

**Review by konangc87 — APPROVED** [review]

File: src/config/wmodules-config.c [hunk 3]
```diff
@@ -100,6 +101,12 @@ int Read_WModule(const OS_XML *xml, xml_node *node, void *d1, void *d2)
             return OS_INVALID;
         }
     }
+    else if (!strcmp(node->values[0], WM_EXECD_CONTEXT.name)) {
```
Comment (Dwordcito) [inline src/config/wmodules-config.c:104]: Hi, this is used?

File: src/logcollector/read_win_event_channel.c [hunk 4]
```diff
@@ -466,9 +466,9 @@ void send_channel_event(EVT_HANDLE evt, os_channel *channel)
         wprovider_name = convert_unix_string(provider_name);
 
         if (wprovider_name && (msg_from_prov = get_message(evt, wprovider_name, EvtFormatMessageEvent)) == NULL) {
-            mterror(WM_LOGCOLLECTOR_LOGTAG, 
-                "Could not get message for (%s)",
-                channel->evt_log);
+            merror(
+                "Could not get message for (%s-%s)",
+                channel->evt_log, wprovider_name);
```
Comment (Dwordcito) [inline src/logcollector/read_win_event_channel.c:471]: Revert this!

File: src/os_execd/execd.h [hunk 5]
```diff
@@ -33,20 +39,107 @@ extern time_t pending_upg;
...
+ *
+ */
+void execd_shutdown();
+
+/**
+ * @brief Gets a pointer to the command name (full path).
+ *
+ * @param name    Command name to get the pointer to.
+ * @param timeout Timeout value to write for the current command.
+ *
```
Comment (Dwordcito) [inline src/os_execd/execd.h:114]: AMAZING!

File: src/shared/wmodules_health_status.c [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        cJSON * messageJSON = cJSON_ParseWithOpts(msg, &jsonErrPtr, 0);
+        if (!messageJSON) {
+            mdebug2("Parse JSON error '%s'", msg);
```
Comment (Dwordcito) [inline src/shared/wmodules_health_status.c:68]: jsonErrPtr could be used here!

**Review by Dwordcito — CHANGES_REQUESTED** [review]

File: src/config/wmodules-config.c (hunk 3 above)
Comment (tinachoB) [reply src/config/wmodules-config.c:104]: Nope. I'll remove this!

File: src/logcollector/read_win_event_channel.c (hunk 4 above)
Comment (tinachoB) [reply src/logcollector/read_win_event_channel.c:471]: Will do.

File: src/shared/wmodules_health_status.c (hunk 6 above)
Comment (tinachoB) [reply src/shared/wmodules_health_status.c:68]: will do.

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**